lib/Reader/readerir.cpp

```diff
@@ -3588,8 +3719,22 @@ IRNode *GenIR::loadField(CORINFO_RESOLVED_TOKEN *ResolvedToken, IRNode *Obj,
   int32_t AccessFlags = ObjIsThis ? CORINFO_ACCESS_THIS : CORINFO_ACCESS_ANY;
   AccessFlags |= CORINFO_ACCESS_GET;
   CORINFO_FIELD_INFO FieldInfo;
+  
   getFieldInfo(ResolvedToken, (CORINFO_ACCESS_FLAGS)AccessFlags, &FieldInfo);
 
+ 
+  if (doSimdIntrinsicOpt()) {
+    Type *ObjType = Obj->getType();
+    auto & Context = LLVMBuilder->getContext();
+
+    if (ObjType->isVectorTy()) {
+      IRNode *Index = (IRNode *)ConstantInt::get(Type::getInt32Ty(Context), FieldInfo.offset * 8 / FloatTy->getScalarSizeInBits());
```

Comment:
I think that we want to base the math of the size of the element type: instead of `FloatTy`, we should be using `ObjType->getVectorElementType()`.